fdbclient/admin_client.go

```diff
@@ -0,0 +1,592 @@
+
+// cliAdminClient provides an implementation of the admin interface using the
+// FDB CLI.
+type cliAdminClient struct {
+	// Cluster is the reference to the cluster model.
+	Cluster *fdbtypes.FoundationDBCluster
+
+	// clusterFilePath is the path to the temp file containing the cluster file
+	// for this session.
+	clusterFilePath string
+}
+
+// NewCliAdminClient generates an Admin client for a cluster
+func NewCliAdminClient(cluster *fdbtypes.FoundationDBCluster, _ client.Client) (controllers.AdminClient, error) {
+	clusterFile, err := ioutil.TempFile("", "")
+	if err != nil {
+		return nil, err
+	}
+	clusterFilePath := clusterFile.Name()
+
+	defer clusterFile.Close()
+	_, err = clusterFile.WriteString(cluster.Status.ConnectionString)
+	if err != nil {
+		return nil, err
+	}
+	err = clusterFile.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	return &cliAdminClient{Cluster: cluster, clusterFilePath: clusterFilePath}, nil
+}
+
+// cliCommand describes a command that we are running against FDB.
+type cliCommand struct {
+	// binary is the binary to run.
+	binary string
+
+	// command is the command to execute.
+	command string
+
+	// version is the version of FoundationDB we should run.
+	version string
+
+	// args provides alternative arguments in place of the exec command.
+	args []string
+}
+
+// hasTimeoutArg determines whether a command accepts a timeout argument.
+func (command cliCommand) hasTimeoutArg() bool {
+	return command.binary == "" || command.binary == "fdbcli"
+}
+
+// hasDashInLogDir determines whether a command has a log-dir argument or a
+// logdir argument.
+func (command cliCommand) hasDashInLogDir() bool {
+	return command.binary == "" || command.binary == "fdbcli"
+}
+
+// getClusterFileFlag gets the flag this command uses for its cluster file
+// argument.
+func (command cliCommand) getClusterFileFlag() string {
+	if command.binary == "fdbrestore" {
+		return "--dest_cluster_file"
+	}
+	return "-C"
+}
+
+// getBinaryPath generates the path to an FDB binary.
+func getBinaryPath(binaryName string, version string) string {
+
+	shortVersion := version[:strings.LastIndex(version, ".")]
+	return fmt.Sprintf("%s/%s/%s", os.Getenv("FDB_BINARY_DIR"), shortVersion, binaryName)
+}
+
+// runCommand executes a command in the CLI.
+func (client *cliAdminClient) runCommand(command cliCommand) (string, error) {
+	version := command.version
+	if version == "" {
+		version = client.Cluster.Status.RunningVersion
+	}
+	if version == "" {
+		version = client.Cluster.Spec.Version
+	}
+
+	binaryName := command.binary
+	if binaryName == "" {
+		binaryName = "fdbcli"
+	}
+
+	binary := getBinaryPath(binaryName, version)
+	hardTimeout := DefaultCLITimeout
+	args := make([]string, 0, 9)
+	args = append(args, command.args...)
+	if len(args) == 0 {
+		args = append(args, "--exec", command.command)
+	}
+
+	args = append(args, command.getClusterFileFlag(), client.clusterFilePath, "--log")
+
+	if binaryName == "fdbcli" {
```

Comment:
Nothing that you have to solve in this PR but I think it makes sense to move that into a method.